### PR TITLE
Get rid of hardcoded jdbc url

### PR DESCRIPTION
**Pull request by dwatl**

### CODE REVIEW

File: src/main/java/com/atlassian/db/replica/internal/aurora/AuroraJdbcUrl.java [hunk 1]
```diff
@@ -0,0 +1,25 @@
+package com.atlassian.db.replica.internal.aurora;
+
+public class AuroraJdbcUrl {
```
Comment (wyrzyk) [inline src/main/java/com/atlassian/db/replica/internal/aurora/AuroraJdbcUrl.java:3]: It's internal but it's a parameter for the API constructor (AuroraCluster). Please make it API or fix the constructor to use API only classes.

File: src/main/java/com/atlassian/db/replica/internal/aurora/AuroraJdbcUrl.java [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    private static final String prefix = "jdbc:postgresql://";
+    private final AuroraEndpoint endpoint;
+    private final String databaseName;
+
+    public AuroraJdbcUrl(AuroraEndpoint endpoint, String databaseName) {
```
Comment (wyrzyk) [inline src/main/java/com/atlassian/db/replica/internal/aurora/AuroraJdbcUrl.java:8]: The same as above. Is `AuroraEndpoint` API or not?

File: src/test/java/com/atlassian/db/replica/it/example/aurora/AuroraClusterTest.java [hunk 3]
```diff
@@ -61,22 +65,22 @@ public void shouldUtilizeReplicaForReadQueriesForSynchronisedWrites() throws SQL
     private SqlCall<Connection> initializeConnectionPool(final DatabaseCall decisionLog) throws SQLException {
         final ReplicaNode replicaNode = new ReplicaNode();
         final ConnectionProvider connectionProvider = new AuroraConnectionProvider(
```
Comment (wyrzyk) [inline src/test/java/com/atlassian/db/replica/it/example/aurora/AuroraClusterTest.java:67]: Should `AuroraConnectionProvider` accept Strings?

File: src/test/java/com/atlassian/db/replica/it/example/aurora/AuroraClusterTest.java [hunk 4]
```diff
@@ -31,8 +33,10 @@
 import static org.assertj.core.api.Assertions.assertThat;
 
 public class AuroraClusterTest {
-    final String writerEndpoint = "jdbc:postgresql://database-1.cluster-crmnlihjxqlm.eu-central-1.rds.amazonaws.com:5432/newdb";
-    final String readerEndpoint = "jdbc:postgresql://database-1.cluster-ro-crmnlihjxqlm.eu-central-1.rds.amazonaws.com:5432/newdb";
+    final AuroraJdbcUrl writerUrl = new AuroraJdbcUrl(
+        AuroraEndpoint.parse("database-1.cluster-crmnlihjxqlm.eu-central-1.rds.amazonaws.com:5432"), "newdb");
```
Comment (wyrzyk) [inline src/test/java/com/atlassian/db/replica/it/example/aurora/AuroraClusterTest.java:37]: Can we avoid static methods in API?

File: src/test/java/com/atlassian/db/replica/it/example/aurora/AuroraClusterTest.java [hunk 5]
```diff
@@ -6,6 +6,8 @@
 import com.atlassian.db.replica.api.aurora.ReplicaNode;
 import com.atlassian.db.replica.api.reason.Reason;
 import com.atlassian.db.replica.api.reason.RouteDecision;
+import com.atlassian.db.replica.internal.aurora.AuroraEndpoint;
```
Comment (wyrzyk) [inline src/test/java/com/atlassian/db/replica/it/example/aurora/AuroraClusterTest.java:9]: This is an example of lib usage. It should avoid calling internals.

File: src/test/java/com/atlassian/db/replica/it/example/aurora/AuroraClusterTest.java [hunk 6]
```diff
@@ -31,8 +33,10 @@
 import static org.assertj.core.api.Assertions.assertThat;
 
 public class AuroraClusterTest {
-    final String writerEndpoint = "jdbc:postgresql://database-1.cluster-crmnlihjxqlm.eu-central-1.rds.amazonaws.com:5432/newdb";
-    final String readerEndpoint = "jdbc:postgresql://database-1.cluster-ro-crmnlihjxqlm.eu-central-1.rds.amazonaws.com:5432/newdb";
+    final AuroraJdbcUrl writerUrl = new AuroraJdbcUrl(
```
Comment (wyrzyk) [inline src/test/java/com/atlassian/db/replica/it/example/aurora/AuroraClusterTest.java:36]: Can it be simplified? (for example, created in one step if we never need the indirect AuroraEndpoint object)

File: src/test/java/com/atlassian/db/replica/it/example/aurora/AuroraClusterTest.java (hunk 4 above)
Comment (dwatl) [reply src/test/java/com/atlassian/db/replica/it/example/aurora/AuroraClusterTest.java:37]: What's wrong with static methods in API?

Comment (wyrzyk) [reply src/test/java/com/atlassian/db/replica/it/example/aurora/AuroraClusterTest.java:37]: Nothing 😉.

File: src/test/java/com/atlassian/db/replica/it/example/aurora/AuroraClusterTest.java (hunk 3 above)
Comment (dwatl) [reply src/test/java/com/atlassian/db/replica/it/example/aurora/AuroraClusterTest.java:67]: I gotta rethink ConnectionProvider, I wonder if it should be more multi-replica friendly

**Pull request merged**